azurerm/internal/services/network/tests/virtual_hub_route_table_test.go

```diff
@@ -0,0 +1,250 @@
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualHubRouteTableExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMVirtualHubRouteTable_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_virtual_hub_route_table", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMVirtualHubRouteTableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMVirtualHubRouteTable_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualHubRouteTableExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMVirtualHubRouteTable_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualHubRouteTableExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMVirtualHubRouteTableExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Network.HubRouteTableClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("network HubRouteTable not found: %s", resourceName)
+		}
+
+		id, err := parse.VirtualHubRouteTableID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.VirtualHubName, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Network HubRouteTable %q does not exist", id.Name)
+			}
+
+			return fmt.Errorf("bad: Get on Network.HubRouteTableClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMVirtualHubRouteTableDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Network.HubRouteTableClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_network_hub_route_table" {
+			continue
+		}
+
+		id, err := parse.VirtualHubRouteTableID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.VirtualHubName, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Get on Network.HubRouteTableClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMVirtualHubRouteTable_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-vhubroutetable-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctest-vnet-%d"
```

Comment:
For all resources that support it can we use capitals for their names please?

Reply:
updated